shared/constants/permissions.ts

```diff
@@ -1,9 +1,11 @@
 export const CaveatTypes = Object.freeze({
   restrictReturnedAccounts: 'restrictReturnedAccounts' as const,
+  restrictNetworkSwitching: 'restrictNetworkSwitching' as const,
 });
 
 export const RestrictedMethods = Object.freeze({
   eth_accounts: 'eth_accounts',
+  wallet_switchEthereumChain: 'wallet_switchEthereumChain',
```

Comment:
Since this is in fact an endowment, I'm not sure that it should be included here.

Reply:
whoops thought I'd removed this.

Reply:
done here: https://github.com/MetaMask/metamask-extension/pull/24415/commits/465ae6bbfdd4a69ef07cf122bac5fb214a4ecd98